### PR TITLE
fix the libtorch version mismatch issue

### PR DESCRIPTION
**Pull request by lanluo-nvidia**
# Description

For the windows build issue, I am able to identify the issue is caused by the torchtrt.dll:
It is complaining the getinstance method in the UndefinedTensorImpl class

The procedure entry point
?getinstance@UndefinedTensorImpl@c10@@CAAEAU12@XZ
could not be located in the dynamic link library
C:\Users\lanl\git\venv_py310\Lib\site-packages\torch_tensorrt\
lib\torchtrt.dll.
I believe this is related to the libtorch version:
currently we use the one from:

https://download.pytorch.org/libtorch/${CHANNEL}/${CU_VERSION}/libtorch-win-shared-with-deps-latest.zip
which is causing the issue.

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: toolchains/ci_workspaces/MODULE.bazel.tmpl [hunk 1]
```diff
@@ -110,17 +89,23 @@ http_archive(
...
+    path = "${TORCH_INSTALL_PATH}",
+    build_file = "third_party/libtorch/BUILD"
+)
 
-#new_local_repository(
-#    name = "libtorch_pre_cxx11_abi",
-#    path = "${TORCH_INSTALL_PATH}",
-#    build_file = "third_party/libtorch/BUILD"
-#
+new_local_repository(
```
Comment (narendasan) [inline toolchains/ci_workspaces/MODULE.bazel.tmpl:98]: Keep in mind if we do this it becomes hard to build the cxx11 abi builds in CI. Consider just replacing the problematic one (libtorch_win) for now. Alternatively, we could replace both `libtorch_win`, and `libtorch_pre_cxx11_abi` but keep `libtorch` as `http_archive`, thats probably best of both worlds for CI

Comment (lanluo-nvidia) [reply toolchains/ci_workspaces/MODULE.bazel.tmpl:105]: Yes, this is a very good point, I have modified as suggested and also enabled release build to see ensure cxx11 abi build is successful in ci.

**Comment by HolyWu** [comment]
I don't think simply changing to local_repository will solve the issue, since the locally installed torch in building wheel phase is still the latest nightly version.

**Comment by lanluo-nvidia** [comment]
> I don't think simply changing to local_repository will solve the issue, since the locally installed torch in building wheel phase is still the latest nightly version.

I have uninstall ed latest nightly version, always use the version we have dependencies on py/requirements.txt
this does no longer show us the error in windows.

However there is a new error in windows for cu118:
 torchvision 0.19.0.dev20240617+cu118 depends on torch==2.5.0.dev20240616+cu118 in windows, 
 however https://download.pytorch.org/whl/nightly/torch/ does not have a 2.5.0.dev20240616+cu118

**Review by narendasan — APPROVED** [review]
LGTM

File: .github/scripts/install-torch-tensorrt.sh [hunk 2]
```diff
@@ -8,6 +8,11 @@ PLATFORM=$(python -c "import sys; print(sys.platform)")
 pip install --pre ${TORCH_TORCHVISION} --index-url ${INDEX_URL}
 pip install --pre -r ${PWD}/tests/py/requirements.txt --use-deprecated legacy-resolver
 
+export CUDA_HOME="$(echo ${CUDA_PATH} | sed -e 's#\\#\/#g')"
+export TORCH_INSTALL_PATH="$(python -c "import torch, os; print(os.path.dirname(torch.__file__))" | sed -e 's#\\#\/#g')"
+
+cat ${PWD}/toolchains/ci_workspaces/MODULE.bazel.tmpl | envsubst > ${PWD}/MODULE.bazel
```
Comment (HolyWu) [inline .github/scripts/install-torch-tensorrt.sh:14]: Was the addition of these three lines necessary? `MODULE.bazel` is only used in build job to create the wheel. But `install-torch-tensorrt.sh` is only used in tests job so `MODULE.bazel` is having no use here.

Comment (lanluo-nvidia) [reply .github/scripts/install-torch-tensorrt.sh]: @HolyWu  thanks for pointing out this, it is indeed not required, removed as suggested.
